src/main/java/org/assertj/core/api/ThrowableAssert.java

```diff
@@ -63,4 +63,11 @@ public static Throwable catchThrowable(ThrowingCallable shouldRaiseThrowable) {
     }
     return null;
   }
+
+  @SuppressWarnings("unchecked")
+  public static <E extends Throwable> E catchThrowableOfType(ThrowingCallable shouldRaiseThrowable, Class<E> type) {
+    Throwable t = AssertionsForClassTypes.catchThrowable(shouldRaiseThrowable);
+    new ThrowableAssert(t).isInstanceOf(type);
```

Comment:
following my other comment, add `.overridingErrorMessage("Expecting code to throw a <%s> but did not", type)` before `.isInstanceOf`

Reply:
Done.